### PR TITLE
[Fixes #5990] revoke owner permissions for resource after approval 

### LINKED ISSUE #5990: Revoke owners right to resource after approval
**Is your feature request related to a problem? Please describe.**
When advanced flow is on id like to disable owners rights to approved resources. In case the owner wants to make a change he has to ask about permission group manager or administrator(depends on flow)

**Describe the solution you'd like**
In case one of the advanced flow is on once resource is approved (by owner/manager/admin) owners permissions should be revoked. Edit button should be no longer available, we should present a new one instead: Request change. This button is presented only to the resource owner. After using it new view is presented. in this view owner can place reasoning why he wants to change something (1000chars). The reasoning is sent to manager/admin (depends on flow) through the messaging system and email notification

### PR DESCRIPTION
**Pull request by petrus7**
FIX #5990 

Manage user rights to the resource, base it on ADMIN_MODERATE_UPLOADS and RESOURCE_PUBLISHING. When one of them is on resource owner has no longer possibility to modify resource after it was approved. He has to ask about permissions unapproving it  

He has to request a change through resource_owner_perm.py

Changes in html views to introduce new button
Add pinax email templates for this functionality

Documentation is here:
https://github.com/GeoNode/documentation/pull/30

<Include a few sentences describing the overall goals for this Pull Request>

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [x] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/base/resource_owner_perm.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from guardian.shortcuts import get_perms, remove_perm, assign_perm
```
Comment (afabiani) [inline geonode/base/resource_owner_perm.py:1]: Missing headers on new files

File: geonode/services/templates/services/service_detail.html [hunk 2]
```diff
@@ -95,23 +96,13 @@ <h3>{% trans "Service Resources" %} <span class="badge">{{ total_resources }}</s
 </div>
 {% endblock %}
 {% block sidebar %}
-
-
-    {% get_obj_perms request.user for service as "resource_perms" %}
```
Comment (afabiani) [inline geonode/services/templates/services/service_detail.html:100]: This change here seems to be outside the scope of the PR.

Which is the intent of this change?

File: geonode/base/models.py [hunk 3]
```diff
@@ -80,6 +80,24 @@
 
 logger = logging.getLogger(__name__)
 
+READ_PERMISSIONS = [
```
Comment (afabiani) [inline geonode/base/models.py:83]: Those arrays are duplicated. They have been already defined in utils.py.
Please do not duplicate code and variables.

File: geonode/base/models.py [hunk 4]
```diff
@@ -540,6 +558,21 @@ class ResourceBase(PolymorphicModel, PermissionLevelMixin, ItemBase):
     """
     Base Resource Object loosely based on ISO 19115:2003
     """
+    BASE_PERMISSIONS = {
```
Comment (afabiani) [inline geonode/base/models.py:561]: Better to define this dictionary outside, similar to the other permissions.

This might be useful in general.

File: geonode/base/resource_owner_perm.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+from django.conf import settings
+
+
+class ManageResourceOwnerPermissions:
```
Comment (afabiani) [inline geonode/base/resource_owner_perm.py:5]: Why we need a brand new Python file dedicated to this class?

Comment (petrus7) [reply geonode/base/resource_owner_perm.py:5]: will be moved to utils

File: geonode/services/templates/services/service_detail.html (hunk 2 above)
Comment (petrus7) [reply geonode/services/templates/services/service_detail.html:100]: At the beggining i thought services also are included in the advanced pubblish workflow. so i did some preparations in this view too. Will restore it to previous state

File: geonode/base/resource_owner_perm.py (hunk 1 above)
Comment (petrus7) [reply geonode/base/resource_owner_perm.py:1]: file will be removed

File: geonode/base/models.py (hunk 3 above)
Comment (petrus7) [reply geonode/base/models.py:83]: right, forgot about this artifact

File: geonode/base/models.py (hunk 4 above)
Comment (petrus7) [reply geonode/base/models.py:561]: My code assumes it is defined on a class level. My code assumes a polymorphic mechanism according to these dictionaries. More to PERMISSIONS dict but BASE_PERMISSIONS are here for consistency

**Comment by petrus7** [comment]
@afabiani  what is wrong with the title? this is exactly what the code do

File: geonode/base/models.py (hunk 4 above)
Comment (afabiani) [reply geonode/base/models.py:561]: ok, so forget my comment
